include/VectorGraph.h

```diff
@@ -0,0 +1,729 @@
+/*
+ * VecorGraph.h - Vector graph widget and model implementation
+ *
+ * Copyright (c) 2024 szeli1 </at/gmail/dot/com> TODO
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LMMS_GUI_VECTORGRAPH_H
+#define LMMS_GUI_VECTORGRAPH_H
+
+#include <vector>
+#include <mutex>
+#include <QPainterPath>
+#include <QWidget>
+#include <QCursor>
+#include <QMenu>
+
+#include "Model.h"
+#include "ModelView.h"
+#include "lmms_basics.h"
+#include "AutomatableModel.h"
+#include "JournallingObject.h"
+
+namespace lmms
+{
+
+class VectorGraphModel;
+class VectorGraphDataArray;
+class FloatModel;
+
+namespace gui
+{
+class LMMS_EXPORT VectorGraphView : public QWidget, public ModelView
+{
+	Q_OBJECT
+	// set default VectorGraph css colors and styles
+	Q_PROPERTY(QColor vectorGraphDefaultAutomatedColor MEMBER m_vectorGraphDefaultAutomatedColor NOTIFY changedDefaultColors)
+	Q_PROPERTY(QColor vectorGraphDefaultLineColor MEMBER m_vectorGraphDefaultLineColor NOTIFY changedDefaultColors)
+	Q_PROPERTY(QColor vectorGraphDefaultActiveColor MEMBER m_vectorGraphDefaultActiveColor NOTIFY changedDefaultColors)
+	Q_PROPERTY(QColor vectorGraphDefaultFillColor MEMBER m_vectorGraphDefaultFillColor NOTIFY changedDefaultColors)
+	Q_PROPERTY(QColor vectorGraphSecondaryLineColor MEMBER m_vectorGraphSecondaryLineColor NOTIFY changedDefaultColors)
+	Q_PROPERTY(QColor vectorGraphSecondaryActiveColor MEMBER m_vectorGraphSecondaryActiveColor NOTIFY changedDefaultColors)
+	Q_PROPERTY(QColor vectorGraphSecondaryFillColor MEMBER m_vectorGraphSecondaryFillColor NOTIFY changedDefaultColors)
+
+	Q_PROPERTY(int fontSize MEMBER m_fontSize)
+public:
+
+	VectorGraphView(QWidget * parentIn, int widthIn, int heightIn,
+		unsigned int pointSizeIn, unsigned int maxLengthIn, bool shouldApplyDefaultVectorGraphColorsIn);
+	~VectorGraphView();
+
+	void setLineColor(QColor colorIn, unsigned int dataArrayLocationIn);
+	void setActiveColor(QColor colorIn, unsigned int dataArrayLocationIn);
+	void setFillColor(QColor colorIn, unsigned int dataArrayLocationIn);
+	void setAutomatedColor(QColor colorIn, unsigned int dataArrayLocationIn);
```

Comment:
It is easier to understand what comes in as input in a function and without using In, variable names sound too general. It would be hard to replace "In" in all (roughly) 100 functions. I can do it if you want, but I would rather not replace those in `VectorGraph`.

Reply:
Input to a function is defined by the arguments, you don't need to say it. The only case where it would make sense to use it is when there are arguments which are `out`.

I would rather we remove `In` everywhere.

Reply:
Resolved.